main.go

```diff
@@ -124,6 +134,39 @@ func main() {
 	close(finishCh)
 }
 
+func retryFailedTests(testRunner runner.Rspec, maxRetries int) int {
+	// Retry failed tests
+	retries := 0
+	for retries < maxRetries {
+		retries++
+		fmt.Printf("Attempt %d of %d to retry failing tests\n", retries, maxRetries)
+
+		cmd, err := testRunner.RetryCommand()
```

Comment:
@nprizal and I talked about unit testing this function by providing a simple command(i.e 'cat banana') to the test runner such that it would be easy to test this func. 
However, the `RetryCommand` will append `--only-failures` to the test command we provide and `cat banana --only-failures` isn't a valid command to run. I didn't add unit tests for this but have done manual testing.

Reply:
You can use `true` as the command rather than `cat banana`, as `true --only-failures` will exit with 0:

```
~/version-control/buildkite/buildkite
❯ true --only-failures

~/version-control/buildkite/buildkite
❯ echo $?
0
```

Reply:
Thanks Steve that's exactly what i'm looking for! I have added the unit tests for this retry logic